### ISSUE TITLE
unable to change the value of subtitle border size to decimals through settings

### ISSUE THREAD

**Issue by chimneysweeper98** [iina/iina#4799]
- macOS 14.2.1
- IINA 1.3.4

**Expected behavior:**
able to set the value of subtitle border size to decimals through settings

**Actual behavior:**
I can only do it with the subtitle panel(see in the first screenshot), but there are only some preset values.
When I do it through settings, it's not working, and it shows no border at all(see in the second screenshot).
It only works when you put in an integer.

<img width="1728" alt="Screenshot 2024-02-02 at 02 31 13" src="https://github.com/iina/iina/assets/158500455/41d65952-63d8-4a60-b12d-721fbf1b79bc">
<img width="1728" alt="Screenshot 2024-02-02 at 02 32 06" src="https://github.com/iina/iina/assets/158500455/f7a4dd08-ef4f-49d0-ae74-4792f1eb9820">

<details>
The subtitle used in the screenshots is an external subtitle(srt).

**Comment by low-batt** [comment]
Thank you very much for reporting this. The clear detailed report is appreciated. Reproduced for me. This is a defect.

Your analysis is correct. The root cause is this line of code in `MPVController.mpvInit` that specifies the type of this setting as an integer:
```swift
setUserOption(PK.subBorderSize, type: .int, forName: MPVOption.Subtitles.subBorderSize)
```

That code is used to apply the values established using IINA's settings. When the subtitles panel is used it goes through a different code path that correctly treats this setting as having a floating point value.

From the `mpv` manual entry for the [sub-border-size](https://mpv.io/manual/stable/#options-sub-border-size) option:
> Size of the sub font border in scaled pixels (see `--sub-font-size` for details). A value of 0 disables borders.

The `0.5` value was rounded down to `0` disabling borders.

The code in `MPVController.mpvInit` is also using the wrong type went setting [sub-font-size](https://mpv.io/manual/stable/#options-sub-font-size):
```swift
setUserOption(PK.subTextSize, type: .int, forName: MPVOption.Subtitles.subFontSize)
```

Also [sub-shadow-offset](https://mpv.io/manual/stable/#options-sub-shadow-offset):
```swift
setUserOption(PK.subShadowSize, type: .int, forName: MPVOption.Subtitles.subShadowOffset)
```

I confirmed `mpv` treats this as a floating point value:
```text
low-batt@gag ~$ echo '{ "command": ["get_property", "sub-shadow-offset"] }' | socat - /tmp/mpv-socket | jq
{
  "data": 0.500000,
  "request_id": 0,
  "error": "success"
}
```
